lib/Yield/HexDump.cpp

```diff
@@ -0,0 +1,302 @@
+/// \file HexDump.cpp
+
+//
+// This file is distributed under the MIT License. See LICENSE.md for details.
+//
+
+#include "boost/icl/interval_map.hpp"
+
+#include "llvm/ADT/StringExtras.h"
+#include "llvm/Support/Error.h"
+#include "llvm/Support/FormatVariadic.h"
+
+#include "revng/EarlyFunctionAnalysis/FunctionMetadataCache.h"
+#include "revng/Model/RawBinaryView.h"
+#include "revng/Pipeline/AllRegistries.h"
+#include "revng/Pipeline/Location.h"
+#include "revng/Pipes/FileContainer.h"
+#include "revng/Pipes/Kinds.h"
+#include "revng/Pipes/Ranks.h"
+#include "revng/Support/IRHelpers.h"
+
+namespace boost::icl {
+
+template<>
+struct is_discrete<MetaAddress> {
+  using type = is_discrete;
+  BOOST_STATIC_CONSTANT(bool, value = true);
+};
+
+template<>
+struct identity_element<MetaAddress> {
+  static MetaAddress value() { return MetaAddress{}; }
+};
+
+template<>
+struct has_difference<MetaAddress> {
+  using type = has_difference;
+  BOOST_STATIC_CONSTANT(bool, value = true);
+};
+
+template<>
+struct difference_type_of<MetaAddress> {
+  using type = uint64_t; // assume we compare only .address() within same type,
+                         // epoch and address space
```

Comment:
1. Capital letter
2. Most comments should take the whole line, no indented comments

Reply:
1. boost expects `type` with lower case.